### PR TITLE
Add tchannel options and callback to forwardToTChannel

### PR DESCRIPTION
**Pull request by davewhat**
forwardToTChannel options for setting target serviceName and cn (caller name)
callback for consumers to receive errors and completion status

### CODE REVIEW

**Comment by Raynos** [comment]
@davewhat make sure to run `npm test` locally to ensure you pass the linter & test suite before code review.

File: node/as/http.js [hunk 1]
```diff
@@ -167,14 +169,15 @@ TChannelHTTP.prototype.forwardToTChannel = function forwardToTChannel(tchannel,
...
+    var options = tchannel.requestOptions(extendInto({
         streamed: true,
         hasNoParent: true
-    });
+    }, tchannelRequestOptions));
     var peer = tchannel.peers.choosePeer(null, options);
     if (!peer) {
         hres.writeHead(503, 'Service Unavailable: no tchannel peer');
         hres.end(); // TODO: error content
+        callback(errors.NoPeerAvailable);
```
Comment (Raynos) [inline node/as/http.js]: `error.NoPeerAvailable()`

**Comment by davewhat** [comment]
@jcorbin for your review.

File: node/test/as-http.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ function allocHTTPBridge(opts) {
     }
 
     function onEgressRequest(hreq, hres) {
-        cluster.asHTTP.forwardToTChannel(cluster.egressChan, hreq, hres);
+        cluster.asHTTP.forwardToTChannel(cluster.egressChan, hreq, hres, {}, function nocallback(err){});
```
Comment (jcorbin) [inline node/test/as-http.js]: isn't that a lint exception? (unused arg err?)  probably at least add `if (err) throw err;` to have it break the test

**Comment by jcorbin** [comment]
instantiate the error as noted by @Raynos , and please also update the http bridge example

**Comment by davewhat** [comment]
had to rebase off of origin/master because python tests were failing.

**Comment by davewhat** [comment]
@jcorbin for your purview.

File: node/test/as-http.js (hunk 2 above)
Comment (jcorbin) [inline node/test/as-http.js]: (bump)?

Comment (davewhat) [inline node/test/as-http.js]: Not sure how I missed that comment.
(shakes fist at github ui)

**Comment by jcorbin** [comment]
:ok: thanks for fixing what will be, if it isn't already, a lint exception

**Pull request merged**